VictoriaAlbert/VictoriaAlbert/MuseumObject.swift

```diff
@@ -0,0 +1,152 @@
+//
+//  MuseumObject.swift
+//  VictoriaAlbert
+//
+//  Created by Sabrina Ip on 11/10/16.
+//  Copyright © 2016 Sabrina. All rights reserved.
+//
+
+import Foundation
+
+internal enum jsonSerialization: Error {
+    case response(jsonData: Any)
+    case records(response: [String : AnyObject])
+}
+
+internal enum recordsParsing: Error {
+    case fields(records: [[String : AnyObject]])
+    case primaryImageId(fields: [String : AnyObject])
+    case objectNumber(fields: [String : AnyObject])
+    case artist(fields: [String : AnyObject])
+    case object(fields: [String : AnyObject])
+    case place(fields: [String : AnyObject])
+    case location(fields: [String : AnyObject])
+    case title(fields: [String : AnyObject])
+    case dateText(fields: [String : AnyObject])
+    case slug(fields: [String : AnyObject])
+    case collectionCode(fields: [String : AnyObject])
+}
+
+class MuseumObject {
+    let primaryImageId: String
+    let objectNumber: String
+    let artist: String
+    let object: String
+    let place: String
+    let location: String
+    let title: String
+    let dateText: String
+    let slug: String
+    let collectionCode: String
+    let smallImage: String
+    let largeImage: String
+    
+    init(primaryImageId : String, objectNumber: String, artist: String, object: String, place: String, location: String, title: String, dateText: String, slug: String, collectionCode: String) {
+        self.primaryImageId = primaryImageId
+        self.objectNumber = objectNumber
+        self.artist = artist
+        self.object = object
+        self.place = place
+        self.location = location
+        self.title = title
+        self.dateText = dateText
+        self.slug = slug
+        self.collectionCode = collectionCode
+        
+        var firstSixChars = ""
+        
+        if primaryImageId.characters.count > 6 {
+            let indexTo6 = primaryImageId.index(primaryImageId.startIndex, offsetBy: 6)
+            firstSixChars = primaryImageId.substring(to: indexTo6)
+        } else {
+            firstSixChars = primaryImageId
+        }
+        
+        self.largeImage = "http://media.vam.ac.uk/media/thira/collection_images/\(firstSixChars)/\(primaryImageId).jpg"
+        self.smallImage = "http://media.vam.ac.uk/media/thira/collection_images/\(firstSixChars)/\(primaryImageId)_jpg_o.jpg"
+    }
+    
+    static func getDataFromJson(data: Data) -> [MuseumObject]? {
+        var museumObject = [MuseumObject]()
+        
+        do {
+            let jsonData = try? JSONSerialization.jsonObject(with: data, options: [])
+            
+            guard let response = jsonData as? [String: AnyObject] else {
+                throw jsonSerialization.response(jsonData: jsonData)
```

Comment:
The reason you're getting a warning on this line is because the ```try?``` is effectively casting the call to ```jsonObject``` as ```Any?```. The solution is that you don't need the ```?``` with ```try``` when it is inside a ```do/catch``` block. Taking it out you get this:

```swift
            let jsonData = try JSONSerialization.jsonObject(with: data, options: [])
            
            guard let response = jsonData as? [String: AnyObject] else {
                throw jsonSerialization.response(jsonData: jsonData)
            }
.
.
.
```
and the error goes away because jsonData is now an ``` Any```, which is what you want anyway. If ```jsonObject``` fails you'll jump to a catch.

Alternatively, btw, if you weren't in a ```do/catch``` and had ```try?``` it'd be in the context of ```if let x = try? y``` which would also give you the non-Optional you want.